### PR TITLE
basic functionality and library tests

### PR DESCRIPTION
**Pull request by GeoffreyEmerson**

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    clientList.init(dummySocket);
+    const guestName = clientList.clientName(dummySocket);
+    assert.equal(guestName,clientList.clientName(dummySocket));
+    clientList.send(dummySocket,'\\nick Bob');
+    assert.notEqual(guestName,clientList.clientName(dummySocket));
+  });
+
+});
+
+class Dummy {
```
Comment (goodwid) [inline test/index.test.js]: Nice mock client..

File: README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+## About
+
+In this lab we create a simple TCP chat server to demonstrate Node's TCP fundamentals.
+
+## Installation
+
+1. Clone this repo to your local drive.
+1. Run `node index.js` to start the server.
+1. Connections to the server can be made with telnet.
+1. Tests can be run with `mocha`.
```
Comment (goodwid) [inline README.md]: This will fail unless you advise users to run npm install first....

Comment (GeoffreyEmerson) [inline README.md]: Testing? Yeah, I guess so. I'll update that.

File: lib/clientList.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+};
+
+exports.length = () => {
+  return clients.length;
+};
+
+function broadcast(self, message) {
+  clients.forEach( clientObj => {
+    if (clientObj.client === self) {
+      clientObj.client.write('\x1b[A\x1b[K' + message);
```
Comment (goodwid) [inline lib/clientList.js]: Nice use of cursor repositioning.  checkout the [ansi-escapes](https://www.npmjs.com/package/ansi-escapes) npm package.

Comment (GeoffreyEmerson) [inline lib/clientList.js]: I was playing around with stuff like that for way, way too long. Eventually abandoned the fancy stuff and went with something that could be turned in.

File: lib/clientList.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    } else {
+      clientObj.client.write(message);
+    }
+  });
+};
+
+function stringCleaning(input) {
+  // remove any unwanted chars here
+  return input.replace(/^\\nick /,'').replace(/[\r|\n]/g,'')
```
Comment (goodwid) [inline lib/clientList.js]: dig the attention to detail.

**Pull request merged**